### PR TITLE
[IOPID-691] : handler logout

### PR DESCRIPTION
**Pull request by pelliccm**
#### List of Changes

Add handler for the logout API

#### Motivation and Context

Permit users to perform a logout

#### How Has This Been Tested?

Unit tests, Postman

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: Logout/handler.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import { IConfig } from "../utils/config";
+import { verifyUserEligibilityMiddleware } from "../utils/middlewares/user-eligibility-middleware";
+
+import { Client } from "../generated/definitions/fast-login/client";
+import {
+  IHslJwtPayloadExtended,
+  hslJwtValidationMiddleware
+} from "../utils/middlewares/hsl-jwt-validation-middleware";
+
+type ILogoutHandler = (
```
Comment (arcogabbo) [inline Logout/handler.ts]: I suggest to rename this with the `T` suffix. the `I` prefix is used only for interfaces
```suggestion
type LogoutHandlerT = (
```

File: Logout/handler.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      flow(
+        E.toError,
+        e =>
+          ResponseErrorInternal(
+            `Something gone wrong calling fast-login: ${e.message}`
+          ),
+        defaultLog.peek.error(e => `${e.detail}`)
+      )
+    ),
+    TE.chainW(
```
Comment (arcogabbo) [inline Logout/handler.ts]: the widen version here is not needed
```suggestion
    TE.chain(
```

File: Logout/handler.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      flow(
+        TE.fromEither,
+        TE.mapLeft(errors =>
+          ResponseErrorInternal(readableReportSimplified(errors))
+        ),
+        defaultLog.taskEither.errorLeft(e => `${e.detail}`),
+        TE.chainW(response => {
```
Comment (arcogabbo) [inline Logout/handler.ts]: may I suggest this? I think it's more readable and we are only using this field below
```suggestion
        TE.chainW(({status}) => {
```

File: Logout/handler.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          if (response.status === 200) {
+            return TE.right(ResponseSuccessJson(void 0));
+          } else {
+            return TE.left<IResponseErrorInternal, IResponseSuccessJson<void>>(
```
Comment (arcogabbo) [inline Logout/handler.ts]: the explicit generic definition is not needed here. do you agree?
```suggestion
            return TE.left(
```

File: Logout/__test__/handler.test.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  spid_level: SpidLevel.L2
+};
+// #endregion
+
+// #region tests
+describe("Logout", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
```
Comment (arcogabbo) [inline Logout/__test__/handler.test.ts:31]: can we also add a test for the error returned by a client case?
I think it should look like this
```ts
  test(`GIVEN a valid user decoded from JWT
    WHEN the client returns an error
      THEN the response should be 500`, async () => {
    const errorStatus = 500;
    logoutMock.mockResolvedValueOnce(E.right({ status: errorStatus }));
    const handler = logoutHandler(fastLoginClientMock);

    const res = await handler(aValidUser);

    expect(logoutMock).toHaveBeenCalledTimes(1);
    expect(logoutMock).toHaveBeenCalledWith({
      body: {
        fiscal_code: aValidUser.fiscal_number
      }
    });
    expect(res).toMatchObject({
      kind: "IResponseErrorInternal",
      detail: expect.stringContaining(`${errorStatus}`)
    });
  });
```

**Review by arcogabbo — COMMENTED** [review]
nice work. just some minor suggestion

**Review by arcogabbo — APPROVED** [review]

File: Logout/handler.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          if (status === 200) {
+            return TE.right(ResponseSuccessJson(void 0));
+          } else {
+            return TE.left(
+              ResponseErrorInternal(
+                `Something gone wrong. Response Status: {${status}}`
+              )
+            );
+          }
+        })
```
Comment (gquadrati) [inline Logout/handler.ts]: Remap status code, as done in session-state handler.

**Pull request merged**